### PR TITLE
Test country-trends.js

### PR DESCRIPTION
**Pull request by ntarn**
Test country-trends.js, add edge case check in case Google Trends API ever breaks in the future and stops providing data

### CODE REVIEW

File: search-sentiments-map/test/country-trends-test.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    const THREE_TIME_RANGE = 3;
+    let currentTime;
+    let mockData;
+    let trendsByCountryThreeHalvesTimeRange;
+    let trendsByCountryThreeTimeRange;
+    let mockThreeHalvesTimeRangeUsResult;
+    let mockThreeTimeRangeUsResult;
+    let mockThreeHalvesTimeRangeUaResult;
+    let mockThreeTimeRangeUaResult;
+    let mockEmptyResult;
```
Comment (carmenBen) [inline search-sentiments-map/test/country-trends-test.js:38]: The only consts/variables that should be declared out here are the ones that are called in the beforeEach block AND an "it" block. If it is only called within one block, it can be declared there instead.

File: search-sentiments-map/test/country-trends-test.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('should get an empty array', async function() {
+      let result = await countryTrends.retrieveCountryTrends('US', 0);
+      assert.deepEqual(result, mockEmpty);
+    });
+  });
+});
```
Comment (carmenBen) [inline search-sentiments-map/test/country-trends-test.js:189]: If you are testing the same function, as the describe function above, this should actually be in the same describe block, but a different it block. And the the description of the it block should be "should return an empty array when there are no trends in the datastore"

File: search-sentiments-map/test/country-trends-test.js [hunk 3]
```diff
@@ -146,42 +112,62 @@ describe('Country Trends', function() {
 
     it('should retrieve the most recent array of US and UA country trends from the datastore beyond the time it takes to retrieve search results', 
         async function() {
+      // Stub calls to the datastore.
+      sinon.stub(Datastore.prototype, 'runQuery').callsFake(() => {
+        return [mockData];
+      });
```
Comment (carmenBen) [inline search-sentiments-map/test/country-trends-test.js:118]: So I think you would want to declare the stub and save it as a variable in the before each, and you could just set the return here.

So for e.g. in the beforeEach have `runQueryStub = sinon.stub(Datastore.prototype, 'runQuery');`
And then within each "it" block `runQueryStub.callsFake(....)`.

On a separate note. Since you aren't actually using the parameters. You may be able to just do `runQueryStub.returns([mockData])`. Also since runQuery should be asynchronous, it may actually need to be `runQueryStub.resolves([mockData])`.

File: search-sentiments-map/test/country-trends-test.js [hunk 4]
```diff
@@ -104,6 +105,7 @@ describe('Country Trends', function() {
         trendsByCountry: trendsByCountryThreeTimeRange,
         globalTrends: [],
       }];
+      runQueryStub = sinon.stub(Datastore.prototype, 'runQuery');
```
Comment (carmenBen) [inline search-sentiments-map/test/country-trends-test.js:108]:
```suggestion

      // Stub calls to the datastore.
      runQueryStub = sinon.stub(Datastore.prototype, 'runQuery');
```

File: search-sentiments-map/routes/country-trends.js [hunk 5]
```diff
@@ -47,6 +47,12 @@ async function retrieveCountryTrends(country, timeRange) {
     descending: true,
   }).filter('timestamp', '<', Date.now() - timeRangeLimit).limit(2);
   const [trendsEntry] = await datastore.runQuery(query);
+  
+  // Check for the chance that Google Trends API breaks and stops providing
+  // data.
```
Comment (hayleejane3) [inline search-sentiments-map/routes/country-trends.js:52]:
```suggestion
  // Handle case where Google Trends API breaks and stops providing
  // data.
```
